File: DFS_BFS/DFS.py
#DFS
visited = [False] * 9

#연결된 노드들에 대한 정보
graph = [
    [],
    [2,3,8],
    [1,7],
    [1,4,5],
    [3,5],
    [3,4],
    [7],
    [2,6,8],
    [1,7]
    
]

def DFS(graph, v, visited):
    # 노드를 방문처리 해준다
    visited[v] = True
    print(v, end =' ')
    
    for i in graph[v]:
        if not visited[i]:
            DFS(graph,i,visited)
            
DFS(graph,1,visited)

File: C++/10주완성/2주차/ dfs.cpp
#include <bits/stdc++.h>
using namespace std;

const int n = 6;
int visited[n];
vector<int> adj[n];

void dfs(int start) {
    visited[start] = true;
    for (int v: adj[start]) {
        if (visited[v] == true) {
            dfs(v);
        }
    }
    return;
}

void main() {
     adj[1].push_back(2); 
     adj[1].push_back(3); 
     adj[2].push_back(4); 
     adj[4].push_back(2); 
     adj[2].push_back(5); 
     dfs(1);
}